libvast/include/vast/plugin.hpp

```diff
@@ -460,6 +460,37 @@ class printer_plugin : public virtual plugin {
   [[nodiscard]] virtual auto printer_allows_joining() const -> bool = 0;
 };
 
+// -- dumper plugin -----------------------------------------------------
+
+/// A dumper plugin transfers a stream of chunks to a sink.
+/// @relates plugin
+class dumper_plugin : public virtual plugin {
+public:
+  // Alias for the byte chunk generation function.
+  using printer
+    = std::function<auto(generator<table_slice>)->generator<chunk_ptr>>;
+
+  // Alias for the byte chunk dumping function.
+  using dumper
+    = std::function<auto(generator<chunk_ptr>)->generator<std::monostate>>;
+
+  /// Returns the dumper.
+  [[nodiscard]] virtual auto
+  make_dumper(const record&, type input_schema, operator_control_plane&) const
```

Comment:
Could we change it to const operator_control_plane& for now? I think it might be good for const correctness.
Later on we might forget that it is non const while we only use const methods anyway.

Reply:
We had a similar discussion [here](https://github.com/tenzir/vast/pull/3009#pullrequestreview-1334983726).